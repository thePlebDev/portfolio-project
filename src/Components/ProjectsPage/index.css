
.node-enter{
  opacity: 0;
}

.node-enter-active{
  opacity:1;
  transition:opacity 1.35s;
}

.node-exit{
  opacity:1;
}

.node-exit-active{
  opacity: 0;
  transition:opacity 1.35s;
}
