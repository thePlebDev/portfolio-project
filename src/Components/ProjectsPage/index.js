import React,{useState,useEffect,useRef} from 'react';
import styled from 'styled-components'

import Project from '../Projects';

import {allData,frontendData,fullStackData} from '../../Assets/Data/projectData.js'


const Container = styled.div`
    position:relative;

`
const ProjectContainer = styled.div`
  width:60%;
  margin:0px auto;
  grid-gap:40px;
  display:grid;
  justify-content:center;
  grid-template-columns:repeat(auto-fit,160px);
  overflow:hidden;
  display:relative;
  padding:80px;

`
const FilterContainer = styled.div`
    width:50%;

    margin:0 auto;
    margin-top:50px;
    display:absolute;
    display:flex;
    justify-content:space-between;
`

const Filter = styled.div`
  font-size:1.6em;
  font-weight:400;
  padding:3px 9px;
  text-transform:uppercase;
  border-bottom:1px solid;
  cursor:pointer;
`


const ProjectsPage =()=>{
  const [state,setState] = useState(allData)


  return(
    <Container>
    <FilterContainer>
      <Filter onClick={()=>setState(allData)}>All</Filter>
      <Filter onClick={()=>setState(fullStackData)}>Full Stack</Filter>
      <Filter onClick={()=>setState(frontendData)}>Frontend</Filter>
    </FilterContainer>
      <ProjectContainer>
        {
          state.map((item)=>{
            return <Project id={item.id} image={item.image}
                       title={item.title} github={item.github}
                       info={item.info} tech={item.tech}/>
          })
        }

      </ProjectContainer>
    </Container>
  )
}

export default ProjectsPage
