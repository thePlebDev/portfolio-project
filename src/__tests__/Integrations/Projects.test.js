
import React from 'react';
import {render} from "@testing-library/react";
import userEvent from '@testing-library/user-event';

import ProjectsPage from '../../Components/ProjectsPage';
//have to test out the projects page.


const allData =[
  {
    title:'all the data',
    id:0
  },
  {
    title:'all the data',
    id:1
  }
]

const frontData =[
  {
    title:'frontend data',
    id:0
  },
  {
    title:'frontend data',
    id:1
  }
]

const fullStackData =[
  {
    title:'fullStack data',
    id:0
  },
  {
    title:'fullStack data',
    id:1
  },
  {
    title:'fullStack data',
    id:2
  }
]



describe('testing the projects page',()=>{
  it('should render properly ',()=>{
    const {asFragment} = render(<ProjectsPage />);
    expect(asFragment()).toMatchSnapshot()
  })

  it('should test the rendered data',()=>{
    const {getAllByText} = render(<ProjectsPage all={allData} />)
    expect(getAllByText('all the data').length).toBe(allData.length)
  })

  it('should test the filter buttons',()=>{
     const {getAllByText,getByText} = render(<ProjectsPage all={allData} front={frontData} full={fullStackData} />)

     //USER CLICKS ON THE FULLSTACK FILTER BUTTON
     userEvent.click(getByText('Full Stack'))
     expect(getAllByText('fullStack data').length).toBe(fullStackData.length)

     //USER CLICKS ON THE FRONT FILTER BUTTON
     userEvent.click(getByText('Frontend'))
     expect(getAllByText('frontend data').length).toBe(frontData.length)

     //USER CLICKS ON THE ALL FILTER BUTTON
     userEvent.click(getByText('All'))
     expect(getAllByText('all the data').length).toBe(allData.lengt)
  })

})
