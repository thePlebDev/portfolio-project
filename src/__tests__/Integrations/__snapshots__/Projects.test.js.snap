// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the projects page should render properly  1`] = `
<DocumentFragment>
  <div
    class="sc-kEqYlL daqxOp"
  >
    <div
      class="sc-crzoUp jMQprb"
    >
      <div
        class="sc-dIsAE hUNgcj"
        data="[object Object],[object Object]"
      >
        All
      </div>
      <div
        class="sc-dIsAE KTHCI"
        data="[object Object]"
      >
        Full Stack
      </div>
      <div
        class="sc-dIsAE KTHCI"
        data="[object Object]"
      >
        Frontend
      </div>
    </div>
    <div
      class="sc-iqAbSa iKllDX"
    >
      <div
        class="sc-pNWxx SIicR"
      >
        <div
          class="sc-jrsJCI iLcdVW"
        >
          View Info
        </div>
        <img
          alt="Dog spotting platform"
          class="sc-fujyUd fHImwe"
          src="pupFinderPortFolio.png"
        />
      </div>
      <div
        class="sc-bdnylx dszVzf"
      >
        <div
          class="sc-gtssRu bUwczj"
        >
          <div
            style="position: relative;"
          >
            <div
              class="sc-dlnjPT eoEyIn"
            >
              Pup Spotter
            </div>
            <div
              class="sc-hKFyIo hFZjWB"
            >
              Dog spotting platform
            </div>
            <div
              class="sc-eCApGN jznTut"
            >
              React / Node / Express / Mongo
            </div>
            <button
              class="sc-jSFkmK Vlhvk"
            >
              <a
                class="sc-iCoHVE laxfsI"
                href="https://github.com/thePlebDev/pupSpotter"
                rel="noopener noreferrer"
                target="_blank"
              >
                View Code
              </a>
            </button>
            <div
              class="sc-gKAblj dKMEDP"
            >
               
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                style="color: rgb(227, 27, 109); font-size: 40px;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-pNWxx SIicR"
      >
        <div
          class="sc-jrsJCI iLcdVW"
        >
          View Info
        </div>
        <img
          alt="Portfolio Page"
          class="sc-fujyUd fHImwe"
          src="portfolioHome.png"
        />
      </div>
      <div
        class="sc-bdnylx dszVzf"
      >
        <div
          class="sc-gtssRu bUwczj"
        >
          <div
            style="position: relative;"
          >
            <div
              class="sc-dlnjPT eoEyIn"
            >
              Personal
            </div>
            <div
              class="sc-hKFyIo hFZjWB"
            >
              Portfolio Page
            </div>
            <div
              class="sc-eCApGN jznTut"
            >
              React / React Testing Library
            </div>
            <button
              class="sc-jSFkmK Vlhvk"
            >
              <a
                class="sc-iCoHVE laxfsI"
                href="https://github.com/thePlebDev/portfolio-project"
                rel="noopener noreferrer"
                target="_blank"
              >
                View Code
              </a>
            </button>
            <div
              class="sc-gKAblj dKMEDP"
            >
               
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                style="color: rgb(227, 27, 109); font-size: 40px;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
